app/src/main/java/com/humberto/tasky/core/domain/agenda/LocalAgendaDataSource.kt

```diff
@@ -9,7 +9,7 @@ import kotlinx.coroutines.flow.Flow
 import java.time.LocalDate
 
 interface LocalAgendaDataSource {
-    fun getAgendaForDate(localDate: LocalDate): Flow<List<AgendaItem>>
+    suspend fun getAgendaForDate(localDate: LocalDate): Flow<List<AgendaItem>>
```

Comment:
When you return a Flow, marking the function as suspending does not make sense. With suspending functions, you want to return a single result, whereas with a Flow, you want multiple results over time.

```
fun getAgendaItems(): Flow<List<Event>>

// Correct usage
getAgendaItems()
  .onEach { /*  process each item */ }
  .launchIn(someScope)
```

Using a suspending function to return a flow is also awkward because you need a coroutine scope to call the function.

```
suspend fun getAgendaItems(): Flow<List<Event>>

// usage
viewModelScope.launch {
  getAgendaItems().collect {
  
  }
}
```